Unsupported field HourOfDay during XML serialization of partial DV_DATE_TIME

When I'm trying to serialize a composition containing a DvDateTime like `2023-01-01` into XML I get the follwing exception:

```
javax.xml.bind.MarshalException
 - with linked exception:
[com.sun.istack.SAXException2: com.sun.xml.bind.api.AccessorException: java.time.temporal.UnsupportedTemporalTypeException: Unsupported field: HourOfDay
com.sun.xml.bind.api.AccessorException: java.time.temporal.UnsupportedTemporalTypeException: Unsupported field: HourOfDay]
        at com.sun.xml.bind.v2.runtime.MarshallerImpl.write(MarshallerImpl.java:301)
        at com.sun.xml.bind.v2.runtime.MarshallerImpl.marshal(MarshallerImpl.java:226)
        at javax.xml.bind.helpers.AbstractMarshallerImpl.marshal(AbstractMarshallerImpl.java:131)
...
Caused by: com.sun.istack.SAXException2: com.sun.xml.bind.api.AccessorException: java.time.temporal.UnsupportedTemporalTypeException: Unsupported field: HourOfDay
com.sun.xml.bind.api.AccessorException: java.time.temporal.UnsupportedTemporalTypeException: Unsupported field: HourOfDay
        at com.sun.xml.bind.v2.runtime.XMLSerializer.reportError(XMLSerializer.java:217)
        at com.sun.xml.bind.v2.runtime.XMLSerializer.reportError(XMLSerializer.java:232)
        at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.serializeBody(ClassBeanInfoImpl.java:339)
        at com.sun.xml.bind.v2.runtime.XMLSerializer.childAsXsiType(XMLSerializer.java:666)
        at com.sun.xml.bind.v2.runtime.property.SingleElementNodeProperty.serializeBody(SingleElementNodeProperty.java:128)
        at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.serializeBody(ClassBeanInfoImpl.java:329)
        at com.sun.xml.bind.v2.runtime.XMLSerializer.childAsXsiType(XMLSerializer.java:666)
        at com.sun.xml.bind.v2.runtime.property.ArrayElementNodeProperty.serializeItem(ArrayElementNodeProperty.java:39)
        at com.sun.xml.bind.v2.runtime.property.ArrayElementProperty.serializeListBody(ArrayElementProperty.java:142)
        at com.sun.xml.bind.v2.runtime.property.ArrayERProperty.serializeBody(ArrayERProperty.java:129)
        at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.serializeBody(ClassBeanInfoImpl.java:329)
        at com.sun.xml.bind.v2.runtime.XMLSerializer.childAsXsiType(XMLSerializer.java:666)
        at com.sun.xml.bind.v2.runtime.property.ArrayElementNodeProperty.serializeItem(ArrayElementNodeProperty.java:39)
        at com.sun.xml.bind.v2.runtime.property.ArrayElementProperty.serializeListBody(ArrayElementProperty.java:142)
        at com.sun.xml.bind.v2.runtime.property.ArrayERProperty.serializeBody(ArrayERProperty.java:129)
        at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.serializeBody(ClassBeanInfoImpl.java:329)
        at com.sun.xml.bind.v2.runtime.XMLSerializer.childAsXsiType(XMLSerializer.java:666)
        at com.sun.xml.bind.v2.runtime.property.SingleElementNodeProperty.serializeBody(SingleElementNodeProperty.java:128)
        at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.serializeBody(ClassBeanInfoImpl.java:329)
        at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.serializeBody(ClassBeanInfoImpl.java:320)
        at com.sun.xml.bind.v2.runtime.XMLSerializer.childAsXsiType(XMLSerializer.java:666)
        at com.sun.xml.bind.v2.runtime.property.ArrayElementNodeProperty.serializeItem(ArrayElementNodeProperty.java:39)
        at com.sun.xml.bind.v2.runtime.property.ArrayElementProperty.serializeListBody(ArrayElementProperty.java:142)
        at com.sun.xml.bind.v2.runtime.property.ArrayERProperty.serializeBody(ArrayERProperty.java:129)
        at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.serializeBody(ClassBeanInfoImpl.java:329)
        at com.sun.xml.bind.v2.runtime.XMLSerializer.childAsSoleContent(XMLSerializer.java:563)
        at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.serializeRoot(ClassBeanInfoImpl.java:310)
        at com.sun.xml.bind.v2.runtime.XMLSerializer.childAsRoot(XMLSerializer.java:464)
        at com.sun.xml.bind.v2.runtime.MarshallerImpl.write(MarshallerImpl.java:298)
        ... 9 more
Caused by: com.sun.xml.bind.api.AccessorException: java.time.temporal.UnsupportedTemporalTypeException: Unsupported field: HourOfDay
        at com.sun.xml.bind.v2.runtime.reflect.AdaptedAccessor.get(AdaptedAccessor.java:48)
        at com.sun.xml.bind.v2.runtime.reflect.TransducedAccessor$CompositeTransducedAccessorImpl.writeLeafElement(TransducedAccessor.java:226)
        at com.sun.xml.bind.v2.runtime.property.SingleElementLeafProperty.serializeBody(SingleElementLeafProperty.java:100)
        at com.sun.xml.bind.v2.runtime.ClassBeanInfoImpl.serializeBody(ClassBeanInfoImpl.java:329)
        ... 35 more
Caused by: java.time.temporal.UnsupportedTemporalTypeException: Unsupported field: HourOfDay
        at java.base/java.time.LocalDate.get0(LocalDate.java:698)
        at java.base/java.time.LocalDate.getLong(LocalDate.java:678)
        at java.base/java.time.format.DateTimePrintContext.getValue(DateTimePrintContext.java:308)
        at java.base/java.time.format.DateTimeFormatterBuilder$NumberPrinterParser.format(DateTimeFormatterBuilder.java:2914)
        at java.base/java.time.format.DateTimeFormatterBuilder$CompositePrinterParser.format(DateTimeFormatterBuilder.java:2529)
        at java.base/java.time.format.DateTimeFormatterBuilder$CompositePrinterParser.format(DateTimeFormatterBuilder.java:2529)
        at java.base/java.time.format.DateTimeFormatter.formatTo(DateTimeFormatter.java:1905)
        at java.base/java.time.format.DateTimeFormatter.format(DateTimeFormatter.java:1879)
        at com.nedap.archie.xml.adapters.DateTimeXmlAdapter.marshal(DateTimeXmlAdapter.java:27)
        at com.nedap.archie.xml.adapters.DateTimeXmlAdapter.marshal(DateTimeXmlAdapter.java:15)
        at com.sun.xml.bind.v2.runtime.reflect.AdaptedAccessor.get(AdaptedAccessor.java:46)
        ... 38 more
```

This looks to me the same as #444, just for XML.

--- a/aom/src/main/java/com/nedap/archie/xml/adapters/DateTimeXmlAdapter.java
+++ b/aom/src/main/java/com/nedap/archie/xml/adapters/DateTimeXmlAdapter.java
@@ -4,9 +4,6 @@ import com.nedap.archie.datetime.DateTimeParsers;
 import com.nedap.archie.datetime.DateTimeSerializerFormatters;
 
 import javax.xml.bind.annotation.adapters.XmlAdapter;
-import java.time.LocalDateTime;
-import java.time.OffsetDateTime;
-import java.time.ZonedDateTime;
 import java.time.temporal.ChronoField;
 import java.time.temporal.TemporalAccessor;
 
@@ -22,14 +19,12 @@ public class DateTimeXmlAdapter extends XmlAdapter<String, TemporalAccessor> {
 
     @Override
     public String marshal(TemporalAccessor value) {
-        if(value instanceof LocalDateTime || value instanceof ZonedDateTime || value instanceof OffsetDateTime) {
-            return value.toString();
-        }
         if(!value.isSupported(ChronoField.HOUR_OF_DAY) &&
                 !value.isSupported(ChronoField.MINUTE_OF_HOUR) &&
                 !value.isSupported(ChronoField.SECOND_OF_MINUTE)) {
             return DateTimeSerializerFormatters.ISO_8601_DATE.format(value);
         }
+
         return DateTimeSerializerFormatters.ISO_8601_DATE_TIME.format(value);
     }
 

--- a/aom/src/main/java/com/nedap/archie/xml/adapters/DateTimeXmlAdapter.java
+++ b/aom/src/main/java/com/nedap/archie/xml/adapters/DateTimeXmlAdapter.java
@@ -7,6 +7,7 @@ import javax.xml.bind.annotation.adapters.XmlAdapter;
 import java.time.LocalDateTime;
 import java.time.OffsetDateTime;
 import java.time.ZonedDateTime;
+import java.time.temporal.ChronoField;
 import java.time.temporal.TemporalAccessor;
 
 /**
@@ -23,6 +24,11 @@ public class DateTimeXmlAdapter extends XmlAdapter<String, TemporalAccessor> {
     public String marshal(TemporalAccessor value) {
         if(value instanceof LocalDateTime || value instanceof ZonedDateTime || value instanceof OffsetDateTime) {
             return value.toString();
+        }
+        if(!value.isSupported(ChronoField.HOUR_OF_DAY) &&
+                !value.isSupported(ChronoField.MINUTE_OF_HOUR) &&
+                !value.isSupported(ChronoField.SECOND_OF_MINUTE)) {
+            return DateTimeSerializerFormatters.ISO_8601_DATE.format(value);
         }
         return DateTimeSerializerFormatters.ISO_8601_DATE_TIME.format(value);
     }

--- a/tools/src/test/java/com/nedap/archie/xml/JAXBRMRoundTripTest.java
+++ b/tools/src/test/java/com/nedap/archie/xml/JAXBRMRoundTripTest.java
@@ -1,7 +1,6 @@
 package com.nedap.archie.xml;
 
 import com.nedap.archie.adlparser.ADLParser;
-import com.nedap.archie.adlparser.modelconstraints.RMConstraintImposer;
 import com.nedap.archie.aom.Archetype;
 import com.nedap.archie.query.RMQueryContext;
 import com.nedap.archie.rm.datastructures.Cluster;
@@ -68,7 +67,7 @@ public class JAXBRMRoundTripTest {
         marshaller.marshal(cluster, writer);
         String xml = writer.toString();
         assertThat(xml, containsString(">12:00<"));
-        assertThat(xml, containsString(">2016-01-01T12:00<"));
+        assertThat(xml, containsString(">2016-01-01T12:00:00<"));
         assertThat(xml, containsString(">2016-01-01<"));
 
         System.out.println(xml);

--- a/tools/src/test/java/com/nedap/archie/xml/JAXBRMTest.java
+++ b/tools/src/test/java/com/nedap/archie/xml/JAXBRMTest.java
@@ -2,6 +2,7 @@ package com.nedap.archie.xml;
 
 import com.nedap.archie.rm.datastructures.Element;
 import com.nedap.archie.rm.datavalues.DvText;
+import com.nedap.archie.rm.datavalues.quantity.datetime.DvDateTime;
 import com.nedap.archie.rm.datavalues.quantity.datetime.DvDuration;
 import org.junit.Test;
 import org.threeten.extra.PeriodDuration;
@@ -10,7 +11,7 @@ import javax.xml.bind.Marshaller;
 import javax.xml.bind.Unmarshaller;
 import java.io.StringReader;
 import java.io.StringWriter;
-import java.time.Period;
+import java.time.*;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
@@ -41,6 +42,61 @@ public class JAXBRMTest {
         marshaller.marshal(element, writer);
         String xml = writer.toString();
         assertTrue(xml, xml.contains("-P10D"));
+    }
+
+    @Test
+    public void serializeDvDateTime() throws Exception {
+        Element fullDateElement = new Element("id6",
+                new DvText("DvDateTime"),
+                new DvDateTime(LocalDateTime.of(2015, 1, 1, 12, 10, 12, 0)));
+        StringWriter writer = new StringWriter();
+        Marshaller marshaller = JAXBUtil.getArchieJAXBContext().createMarshaller();
+        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+        marshaller.marshal(fullDateElement, writer);
+        String xml = writer.toString();
+        assertTrue(xml.contains("2015-01-01T12:10:12"));
+        Unmarshaller unmarshaller = JAXBUtil.getArchieJAXBContext().createUnmarshaller();
+        Element unmarshalled = (Element) unmarshaller.unmarshal(new StringReader(xml));
+        assertEquals(fullDateElement, unmarshalled);
+
+        Element yearMonthDayelement = new Element("id6",
+                new DvText("DvDateTime"),
+                new DvDateTime(LocalDate.of(2015, 1, 1)));
+        writer = new StringWriter();
+        marshaller = JAXBUtil.getArchieJAXBContext().createMarshaller();
+        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+        marshaller.marshal(yearMonthDayelement, writer);
+        xml = writer.toString();
+        assertTrue(xml.contains("2015-01-01"));
+        unmarshaller = JAXBUtil.getArchieJAXBContext().createUnmarshaller();
+        unmarshalled = (Element) unmarshaller.unmarshal(new StringReader(xml));
+        assertEquals(yearMonthDayelement, unmarshalled);
+
+        Element yearMonthElement = new Element("id6",
+                new DvText("DvDateTime"),
+                new DvDateTime(YearMonth.of(2015, 1)));
+        writer = new StringWriter();
+        marshaller = JAXBUtil.getArchieJAXBContext().createMarshaller();
+        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+        marshaller.marshal(yearMonthElement, writer);
+        xml = writer.toString();
+        assertTrue(xml.contains("2015-01"));
+        unmarshaller = JAXBUtil.getArchieJAXBContext().createUnmarshaller();
+        unmarshalled = (Element) unmarshaller.unmarshal(new StringReader(xml));
+        assertEquals(yearMonthElement, unmarshalled);
+
+        Element yearElement = new Element("id6",
+                new DvText("DvDateTime"),
+                new DvDateTime(Year.of(2015)));
+        writer = new StringWriter();
+        marshaller = JAXBUtil.getArchieJAXBContext().createMarshaller();
+        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+        marshaller.marshal(yearElement, writer);
+        xml = writer.toString();
+        assertTrue(xml.contains("2015"));
+        unmarshaller = JAXBUtil.getArchieJAXBContext().createUnmarshaller();
+        unmarshalled = (Element) unmarshaller.unmarshal(new StringReader(xml));
+        assertEquals(yearElement, unmarshalled);
     }
 
     @Test

